File: SaltyChat.Client/source/Models/SaltyChat/Configuration.ts
import {RadioTower} from "./RadioTower";


export class Configuration {
    public serverIdentifier: string;
    public teamSpeakName: string;
    public soundPack: string;
    public ingameChannel: number;
    public ingameChannelPassword: string;
    public swissChannels: number[];
    public voiceRanges: number[];
    public radioTowers: RadioTower[];
    public requestTalkStates: boolean;
    public requestRadioTrafficStates: boolean;
    public radioRangeUltraShort: number;
    public radioRangeShort: number;
    public radioRangeLong: number;
}

File: SaltyChat.Server/Models/Configuration.cs
namespace SaltyChat.Server.Models
{
    public class Configuration
    {
        public string ServerIdentifier { get; set; }
        public string SoundPack { get; set; }
        public uint IngameChannel { get; set; }
        public string IngameChannelPassword { get; set; }
        public uint[] SwissChannels { get; set; }
        public float[] VoiceRanges { get; set; }
        public RadioTower[] RadioTowers { get; set; }
        public bool RequestTalkStates { get; set; }
        public bool RequestRadioTrafficStates { get; set; }
        public string MinimumPluginVersion { get; set; }
        public float MegaphoneRange { get; set; }
        public string NamePattern { get; set; }
        public float RadioRangeUltraShort { get; set; }
        public float RadioRangeShort { get; set; }
        public float RadioRangeLong { get; set; }
    }
}